controllers/dataplane/openstackdataplanedeployment_controller.go

```diff
@@ -426,7 +428,11 @@ func (r *OpenStackDataPlaneDeploymentReconciler) setHashes(
 // SetupWithManager sets up the controller with the Manager.
 func (r *OpenStackDataPlaneDeploymentReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&dataplanev1.OpenStackDataPlaneDeployment{}).
+		For(&dataplanev1.OpenStackDataPlaneDeployment{},
+			builder.WithPredicates(predicate.Or(
+				predicate.GenerationChangedPredicate{},
+				predicate.AnnotationChangedPredicate{},
+				predicate.LabelChangedPredicate{}))).
```

Comment:
If nobody else sets the status of the Deployment just the deployment controller itself then not reconciling on status change is OK. (In k8s in general it is allowed to update the status of another resources)

GeneratrionChangePredicate covers the Spec change, and the the other two covers for label and annotation change. Do we need to cover other metadata fields changes? E.g. finalizers, deletionTimestamp?

Reply:
> If nobody else sets the status of the Deployment just the deployment controller itself then not reconciling on status change is OK

No, we don't set status from another controller, which is a bad design though allowed in k8s. 

> Do we need to cover other metadata fields changes? E.g. finalizers, deletionTimestamp?

We don't have finalizers set on the nodeset/deployment resources.  As per the question on reconciling when deletionTimestamp changes, firstly there is no delete reconcile for these CRs and I don't know if there is a need to reconcile when deletionTimestamp changes, as it's for k8s to garbage collect the resource once the finalizers are removed if there is deletionTimestamp.

Reply:
Also what's the scenario you're thinking of that would require reconciliation when finalizers are changed on a CR?

Reply:
To be able to reconcile on delete the controller needs to react on the deletionTimestamp change as that is the only changed field from k8s when the delete is requested. But you said that there is no need to reconcile on delete in these resources so this is not a problem.

Also reconciling on finalizer changes is a moot point if we never put finalizers on these resources.

Reply:
I vaguely remember that we need to look at finalizers during cleanup not to destroy backends while other resources finalizers are still present. In this scenario reacting on a finalizer removal is needed to do the last reconcile on deletion to trigger.

Reply:
I think this case was that I remembered about reconciling on finalizer change https://github.com/openstack-k8s-operators/mariadb-operator/blob/8b86dae4532b4bc9239c4da6425c009b60e24ed6/controllers/mariadbaccount_controller.go#L413

Reply:
oh wait, I think deletionTimeStamp changes generation https://github.com/kubernetes/kubernetes/issues/24935

Reply:
It seems that's also true for finalizers as well from some references, though I'm still to validate that in code.

Reply:
Thanks for the link. That eas new to me that k8s bumps generation for deletedTimestamp.

Reply:
I think all my question is answered here and none of those discussion lead to discovering a real error scenario. so for me this looks OK.